File: justmeet_backend/src/main/java/com/diegodiome/justmeet_backend/model/Comment.java

package com.diegodiome.justmeet_backend.model;

import java.util.Date;

import javax.persistence.Id;
import javax.validation.constraints.NotNull;

import com.fasterxml.jackson.annotation.JsonFormat;

public class Comment {

    @Id 
    private String commentId;

    @NotNull
    private String commentCreator;

    @NotNull
    private String commentBody;

    @NotNull
    @JsonFormat(pattern = "yyyy-MM-dd HH:mm:SS")
    private Date commentDate;

    @NotNull
    private String eventId;

    public String getCommentId() {
        return commentId;
    }

    public void setCommentId(String commentId) {
        this.commentId = commentId;
    }

    public String getCommentCreator() {
        return commentCreator;
    }

    public void setCommentCreator(String commentCreator) {
        this.commentCreator = commentCreator;
    }

    public String getCommentBody() {
        return commentBody;
    }

    public void setCommentBody(String commentBody) {
        this.commentBody = commentBody;
    }

    public String getEventId() {
        return eventId;
    }

    public void setEventId(String eventId) {
        this.eventId = eventId;
    }

    public Date getCommentDate() {
        return commentDate;
    }

    public void setCommentDate(Date commentDate) {
        this.commentDate = commentDate;
    }
}

File: justmeet_backend/src/main/kotlin/com/diegodiome/justmeet_backend/model/Comment.kt
package com.diegodiome.justmeet_backend.model

import com.fasterxml.jackson.annotation.JsonFormat
import java.util.*

data class Comment(
        val commentId: String? = UUID.randomUUID().toString(),
        val commentCreatorId: String,
        var commentCreator: User? = null,
        val commentBody: String,
        val eventId: String,
        @JsonFormat(pattern = "yyyy-MM-dd HH:mm") val commentDate: Date
) {
    constructor() : this("","", User(),"","",Date())
}